app/middleware/services/groups_service.rb

```diff
@@ -0,0 +1,66 @@
+module Services
+  class GroupsService < BaseService
+
+    def fetch
+      if params[:id].present?
+        collection.where(id: params[:id])
+      else
+        collection
+      end
+    end
+
+    def create
+      group = user.groups.new params
+      if group.save
+        { group: group, errors: [] }
+      else
+        { group: nil, errors: group.errors.full_messages }
+      end
+    end
+
+    def update
+      if object.update params
+        { group: object, errors: [] }
+      else
+        { group: nil, errors: object.errors.full_messages }
+      end
+    end
+
+    def delete
+      if object.archive
```

Comment:
Ah..! I think it's better to do logical delete (soft delete) instead of physical delete. The reason is that sometimes users accidentally delete it. Deleting it physically is irreversible. If we don't care about this case, it can be changed to simply delete physically.